src/NumberTheory/vinberg.jl

```diff
@@ -353,3 +353,163 @@ function vinberg_algorithm(S::ZZLat, upper_bound; v0=QQ[0;]::QQMatrix, root_leng
   return [r * B for r in roots]
 end
 
+
+function _get_tau(f::QQMatrix, Qb) ::QQBarFieldElem
+  tau = QQ(0)
+  for lambda in eigenvalues(Qb, f)
+    if abs(lambda)>tau
+      tau = abs(lambda)
+    end
+  end
+  return tau
+end
+
+function _get_bilinealform(L::ZZLat, Qb)
+  #return (a, b)->inner_product(ambient_space(L), a,b)[1,1] -> it doesn't go well as there is no convienient way to convert ZZLat in QQ field to the Qb field
+  return (a,b)-> ((change_base_ring(Qb, a))*change_base_ring(Qb, gram_matrix(ambient_space(L)))*transpose(change_base_ring(Qb, b)))[1]
+end
+
+function _get_eigenvector(f::QQMatrix, lambda::QQBarFieldElem)
+  return eigenspace(f, lambda; side=:left)
+end
+
+function _get_C0(Lf::ZZLatWithIsom, tau::QQBarFieldElem)::PolyRingElem
+  charPolyF = characteristic_polynomial(Lf)
+  x = gen(parent(charPolyF))
+  (n, remainder) = remove(charPolyF, x-1) # remove all x-1 factors
+  return div(remainder, minpoly(parent(charPolyF), tau)) #remove Salem polynomial of salem number tau
+end
+
+function _get_Cfancy(Lf::ZZLatWithIsom, C0)
+  return short_vectors_iterator(lattice(kernel_lattice(Lf, C0)), 2 , 2)
+end
+
+
+@doc raw"""
+  _get_h(L::ZZLat, v, w, Qb, bi_form)->QQMatrix
```

Comment:
This first line in the docstring should be indented by exactly 4 spaces to ensure it is rendered as we want it (namely as code) in the manual)
```suggestion
    _get_h(L::ZZLat, v, w, Qb, bi_form) -> QQMatrix
```